vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java

```diff
@@ -76,25 +51,24 @@ public Progress process(Processing processing) {
                 DocumentPut put = (DocumentPut) op;
                 Document document = put.getDocument();
                 if (document.getDataType().isA(MUSIC_DOCUMENT_TYPE)) {
-
-                    // Set lyrics if already looked up, and set DONE
-                    FieldValue lyrics = (FieldValue)processing.getVariable(VAR_LYRICS);
-                    if (lyrics != null) {
-                        document.setFieldValue("lyrics", lyrics);
-                        logger.info("  Set lyrics, Progress.DONE");
-                        return DocumentProcessor.Progress.DONE;
-                    }
-
-                    // Make a lyric request if not already pending
-                    if (processing.getVariable(VAR_REQ_ID) == null) {
+                    if (processing.getVariable(VAR_REQ_ID) != null) { // A request has been made, check for response
```

Comment:
Long requestId = processing.getVariable(VAR_REQ_ID);
if (requestId != null)
Then you avoid double lookup in concurrent map. I do not think the JIT compiler can optimize that one away.
It also makes i clearer.

Reply:
good idea, fixed!